lib/puppet/provider/hocon_setting/ruby.rb

```diff
@@ -26,9 +22,7 @@ def exists?
       return false
     end
 
-    if type == 'array_element'
-      return Array(@resource[:value]).any? { |v| value.flatten.include?(v) }
-    end
+    return Array(@resource[:value]).any? { |v| value.flatten.include?(v) } if type == 'array_element'
```

Comment:
I wonder if performance wise you want to pull `value.flatten` out of the loop
```suggestion
    if type == 'array_element'
      value_flattened = value.flatten
      return Array(@resource[:value]).any? { |v| value_flattened.include?(v) }
    end
```